lib/ansible/modules/files/jsonpatch.py

```diff
@@ -0,0 +1,455 @@
+        """Perform a 'move' operation."""
+        new_obj, removed = self.remove(from_path, obj)
+        new_obj, chg = self.add(path, removed, new_obj)
+        return new_obj, chg
+
+    def copy(self, from_path, path, obj, **discard):
+        """Perform a 'copy' operation."""
+        value = self._get(from_path, obj)
+        new_obj, chg = self.add(path, value, obj)
+        return new_obj, chg
+
+    def test(self, path, value, obj, **discard):
+        """Perform a 'test' operation.
+
+        This operation supports an additional feature not outlined in
+        RFC 6901 (https://tools.ietf.org/html/rfc6901): The ability to
+        reference every member of an array by using an asterisk (*).
+
+        In such a case, each member of the array at that point in the
+        path will be tested sequentially for the given value, and if
+        a matching value is found, the method will return immediately
+        with a True value.
+
+        Example:
+            {"op": "test", "path": "/array/*/member/property", "value": 2}
+
+        If the object to be tested looked like...
+        {
+            "array": [
+                {"member": {"property": 1}},
+                {"member": {"property": 2}}
+            ]
+        }
+        ... the result would be True, because an object exists within
+        "array" that has the matching path and value.
+        """
+        elements = path.lstrip('/').split('/')
+        next_obj = obj
+        for idx, elem in enumerate(elements):
+            if elem == "*":  # wildcard
+                if not isinstance(next_obj, list):
+                    raise PathError("'*' does not refer to a JSON array")
+                for sub_obj in next_obj:
+                    dummy, found = self.test('/'.join(elements[(idx + 1):]), value, sub_obj)
+                    if found:
+                        return obj, found
+                return obj, False
+            else:
+                try:
+                    next_obj = next_obj[elem]
+                except KeyError:
+                    raise PathError("'%s' was not found in the JSON object" % elem)
+                except TypeError:  # it's a list
+                    if not elem.isdigit():
+                        raise PathError("'%s' is not a valid index for a JSON array" % elem)
+                    try:
+                        next_obj = next_obj[int(elem)]
+                    except IndexError:
+                        raise PathError("specified index '%s' was not found in JSON array" % elem)
+        return obj, next_obj == value
+
+
+def main():
+    # Parsing argument file
+    module = basic.AnsibleModule(
+        argument_spec=dict(
+            src=dict(required=True, type='str'),
+            dest=dict(required=False, type='str'),
+            operations=dict(required=True, type='list'),
+            backup=dict(required=False, default=False, type='bool'),
+        ),
+        supports_check_mode=False
```

Comment:
How hard would it be to add check_mode support?

Reply:
Not sure, since the RFC doesn't do anything like this (and `test` has rather different behavior), but I imagine it wouldn't be terribly hard.